actions/login-to-gcs/action.yaml

```
@@ -58,20 +63,27 @@ runs:
         fi
         echo "bucket=${BUCKET}" | tee -a ${GITHUB_OUTPUT}
 
-        # Construct service account
-        if [[ "${SERVICE_ACCOUNT}" == "" ]]; then
-          SERVICE_ACCOUNT="github-${{ github.repository_id }}-${ENVIRONMENT}-gcs@grafanalabs-workload-identity.iam.gserviceaccount.com"
+        # set up a service account only if we're NOT using WIF auth.
+        if [[ "${USE_WIF_AUTH}" != "true" ]]; then
+          # Construct service account
+          if [[ "${SERVICE_ACCOUNT}" == "" ]]; then
+            SERVICE_ACCOUNT="github-${{ github.repository_id }}-${ENVIRONMENT}-gcs@grafanalabs-workload-identity.iam.gserviceaccount.com"
+          else
+            SERVICE_ACCOUNT="${SERVICE_ACCOUNT}"
+          fi
+          echo "service_account=${SERVICE_ACCOUNT}" | tee -a ${GITHUB_OUTPUT}
         else
-          SERVICE_ACCOUNT="${SERVICE_ACCOUNT}"
+          echo "service_account=" | tee -a ${GITHUB_OUTPUT}
         fi
-        echo "service_account=${SERVICE_ACCOUNT}" | tee -a ${GITHUB_OUTPUT}
       env:
         BUCKET: ${{ inputs.bucket }}
         ENVIRONMENT: ${{ inputs.environment }}
         SERVICE_ACCOUNT: ${{ inputs.service_account }}
     - uses: google-github-actions/auth@7c6bc770dae815cd3e89ee6cdf493a5fab2cc093 # v3.0.0
+        USE_WIF_AUTH: ${{ inputs.use_wif_auth }}
```

I believe this environment variable goes on the previous step? This should fix the issue with the failing checks.
```suggestion
        USE_WIF_AUTH: ${{ inputs.use_wif_auth }}
    - uses: google-github-actions/auth@7c6bc770dae815cd3e89ee6cdf493a5fab2cc093 # v3.0.0
```

---

Thanks, applied that in the latest commit.